R/fitNullModel.R

```diff
@@ -101,14 +113,30 @@ setMethod("fitNullModel",
 nullModelInvNorm <- function(null.model, cov.mat = NULL,
                              norm.option = c("by.group", "all"),
                              rescale = c("none", "model", "residSD"),
-                             AIREML.tol = 1e-4, 
+                             AIREML.tol = 1e-4,
                              max.iter = 100, EM.iter = 0,
                              verbose = TRUE) {
 
-    updateNullModOutcome(null.model, covMatList=cov.mat, rankNorm.option=norm.option,
-                         rescale=rescale, AIREML.tol=AIREML.tol, 
+    # Update null model format.
+    null.model <- .updateNullModelFormat(null.model)
+
+    new.nullmod <- updateNullModOutcome(null.model, covMatList=cov.mat, rankNorm.option=norm.option,
+                         rescale=rescale, AIREML.tol=AIREML.tol,
                          max.iter=max.iter, EM.iter=EM.iter,
                          verbose=verbose)
+
+    # Add sample id.
+    new.nullmod$fit$sample.id <- null.model$fit$sample.id
```

Comment:
What if it's a nullmod where the `sample.id` are the rownames and not a column?

Reply:
In that case, the `sample.id` column won't exist even after running this command. Because `null.model$fit$sample.id` is `NULL`, it sets `new.nullmod$fit$sample.id` to `NULL`, which means it's not in the new `fit` data frame. I'll add a comment explaining this.

Reply:
`rownames` are already set by the `updateNullModOutcome` function, because it calls `fitNullModel`.